site/docs/v1/tech/lambdas/client-credentials-jwt-populate.adoc

```diff
@@ -0,0 +1,84 @@
+---
+layout: doc
+title: Client Credentials JWT Populate lambda
+description: An overview of the Client Credentials JWT Populate lambda
+---
+
+:sectnumlevels: 0
+
+== Client Credentials JWT Populate lambda
+
+If you would like to augment the claims provided in the JWT before it has been signed you can specify a lambda in the JWT configuration. This lambda will be invoked prior to the token being signed and issued to a user.
+
+When you create a new lambda using the FusionAuth UI we will provide you an empty function for you to implement. If you are using the API to create the lambda you will need to ensure your function has the following signature:
+
+[source,javascript]
+----
+function populate(jwt, recipientEntity, targetEntities, permissions) {
+  // Lambda code goes here
+}
+----
+
+This lambda must contain a function named `populate` that takes four parameters. The parameters that the lambda is passed are:
+
+* `jwt` - the claims object to be signed and return as the JWT payload
+* `permissions` - the permissions assigned to the Entity
+* `recipientEntity` - the Recipient Entity
+* `targetEntities` - the Target Entities
```

Comment:
These are not in the same order as the signature shown above. Should we describe the data type of each of these? The JWT we mention is an object, are the others an array or an object, and what do they contain?

Reply:
Yes, I think that it might be good to describe.  I wish there was a way to print each object (like the test button that we have on webhooks).  This might be more challenging, however.  I will review src/test/java/io/fusionauth/app/action/oauth2/TokenActionClientCredentialsGrantTest.java to see what is returned 🙂

Reply:
You could enable debug and console.log each of these during a client credentials grant :)

Reply:
Sweet!  Followed your and @robotdan's advice and added some json objects.